model/task_lifecycle.go

```diff
@@ -1643,18 +1643,27 @@ func ClearAndResetStrandedHostTask(h *host.Host) error {
 	return nil
 }
 
-// ResetStaleTask fixes a task that has exceeded the heartbeat timeout by
-// marking the current task execution as finished and, if possible, a new
-// execution is created to restart the task.
+// ResetStaleTask fixes a task that has either exceeded the heartbeat timeout
+// or has been marked as aborted but was never ended by the agent.
+// The current task execution is marked as finished and, if the task is not
+// aborted, a new execution is created to restart the task.
 func ResetStaleTask(t *task.Task) error {
 	CheckAndBlockSingleHostTaskGroup(t, t.Status)
 
-	if err := resetSystemFailedTask(t, evergreen.TaskDescriptionHeartbeat); err != nil {
-		return errors.Wrap(err, "resetting task")
+	failureDesc := evergreen.TaskDescriptionHeartbeat
+	if t.Aborted {
+		failureDesc = evergreen.TaskDescriptionAborted
+		if err := t.MarkSystemFailed(failureDesc); err != nil {
+			return errors.Wrap(err, "marking aborted task as system failed")
+		}
+	} else {
+		if err := resetSystemFailedTask(t, failureDesc); err != nil {
+			return errors.Wrap(err, "resetting heartbeat task")
+		}
 	}
 
 	grip.Info(message.Fields{
-		"message":            "successfully fixed stale heartbeat task",
+		"message":            fmt.Sprintf("successfully fixed stale %s task", failureDesc),
```

Comment:
Can we include the failure description a separate field to make Splunk searching easier?